app/src/main/java/com/codepath/chefster/MainActivity.java

```diff
@@ -0,0 +1,81 @@
+package com.codepath.chefster;
+
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class MainActivity extends AppCompatActivity implements GoogleApiClient.OnConnectionFailedListener {
+    private static final String ANONYMOUS = "anonymousUser";
+    // Firebase instance variables
+    private FirebaseAuth firebaseAuth;
+    private FirebaseUser firebaseUser;
+    private String mUsername;
+    private String mPhotoUrl;
+    private GoogleApiClient mGoogleApiClient;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mGoogleApiClient = new GoogleApiClient.Builder(this)
```

Comment:
why do we need the GoogleApiClient in the main Activity, 
I think for signOut we can use FirebaseAuth.getInstance().signOut();